design/sketch.webidl

```diff
@@ -376,12 +430,71 @@ enum GPUIndexFormat {
     "uint32"
 };
 
+// Vertex formats
+// The name of the format specifies the data type of the component, the number of
+// values, and whether the data is normalized.
+//     uchar = unsigned 8-bit value
+//     char = signed 8-bit value
+//     ushort = unsigned 16-bit value
+//     short = signed 16-bit value
+//     half = half-precision 16-bit floating point value
+//     float = 32-bit floating point value
+//     uint = unsigned 32-bit integer value
+//     int = signed 32-bit integer value
+// If no number of values is given in the name, a single value is provided.
+// If the format has the "-bgra" suffix, it means the values are arranged as
+// blue, green, red and alpha values.
+
 enum GPUVertexFormat {
-    "float-r32g32b32a32",
-    "float-r32g32b32",
-    "float-r32g32",
-    "float-r32"
-    // TODO other vertex formats
+    "uchar",
+    "uchar2",
+    "uchar3",
+    "uchar4",
+    "char",
+    "char2",
+    "char3",
+    "char4",
+    "ucharnorm",
```

Comment:
merging words together here doesn't look particularly good
maybe we can have `uchar1norm` instead?

Reply:
Bikeshedding, but I don't like how the "u" and "norm" are separated either.

Reply:
the "norm" used in vertex formats doesn't even match the texture format conventions, where only "unorm" and "snorm" are allowed. OTOH, I don't feel like bikeshedding too much...

Reply:
(same with "int" "short" "char" - above they would be "sint" "sshort" "schar", which are names I definitely don't love. Plus "int" especially has a long history of having variable size....)